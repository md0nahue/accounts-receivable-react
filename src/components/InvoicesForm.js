import React, { useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { addInvoice } from "../actions";

function mapDispatchToProps(dispatch) {
  return {
    addInvoice: invoice => {
      dispatch(addInvoice(invoice));
    }
  };
}

function InvoicesForm({ addInvoice }) {
  let [client, setClient] = useState("");
  let [poNumber, setPoNumber] = useState("");

  const onSubmitHandler = e => {
    e.preventDefault();
    addInvoice({ client, poNumber });
  };

  return (
    <form onSubmit={onSubmitHandler}>
      <div className="form-group">
        <label>Client: </label>
        <select name="client" value={client} onChange={e => setClient(e.target.value)} class="form-control">
          <option value="CircleCi">CircleCi</option>
          <option value="New Relic">New Relic</option>
        </select>
      </div>
      <div className="form-group">
        <label>Purchase Order Number: </label>
        <input
          type="text"
          name="ponumber"
          value={poNumber}
          onChange={e => setPoNumber(e.target.value)}
        />
      </div>
      <button type="submit" className="btn btn-primary">Save</button>
    </form>
  );
}

InvoicesForm.propTypes = {
  addInvoice: PropTypes.func.isRequired
};

export default connect(
  null,
  mapDispatchToProps
)(InvoicesForm);
