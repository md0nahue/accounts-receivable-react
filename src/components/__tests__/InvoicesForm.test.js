import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "jest-dom/extend-expect";

import InvoicesForm from "../InvoicesForm";

const middlewares = [];
const mockStore = configureStore(middlewares);

beforeEach(cleanup);

function renderWithStore(store) {
  return render(
    <Provider store={store}>
      <InvoicesForm />
    </Provider>
  );
}

describe("InvoicesForm Component", () => {
  it("Renders a form matching snapshot", () => {
    const store = mockStore({});
    const { container } = renderWithStore(store);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Dispatches action on "save" click', () => {
    const store = mockStore({});
    const { getByText } = renderWithStore(store);

    fireEvent.click(getByText("Save"));
    const actions = store.getActions();

    expect(actions.length).toEqual(1);
    expect(actions[0].type).toEqual("ADD_INVOICE");
  });
});
