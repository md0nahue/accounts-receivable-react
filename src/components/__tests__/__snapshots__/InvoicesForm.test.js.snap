// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoicesForm Component Renders a form matching snapshot 1`] = `
<form>
  <div
    class="form-group"
  >
    <label>
      Client: 
    </label>
    <select
      class="form-control"
      name="client"
    >
      <option
        value="CircleCi"
      >
        CircleCi
      </option>
      <option
        value="New Relic"
      >
        New Relic
      </option>
    </select>
  </div>
  <div
    class="form-group"
  >
    <label>
      Purchase Order Number: 
    </label>
    <input
      name="ponumber"
      type="text"
      value=""
    />
  </div>
  <button
    class="btn btn-primary"
    type="submit"
  >
    Save
  </button>
</form>
`;
